### PR TITLE
[flutter_releases] Flutter beta 2.12.0-4.1.pre Framework Cherrypicks

### PR DESCRIPTION
**Pull request by itsjustkevin**
# Flutter beta 2.12.0-4.1.pre Framework

## Scheduled Cherrypicks
* None

### CODE REVIEW

**Review by CaseyHillers — APPROVED** [review]

File: bin/internal/release-candidate-branch.version [hunk 1]
```diff
@@ -0,0 +1 @@
+2.12.0-4.1.pre
```
Comment (itsjustkevin) [inline bin/internal/release-candidate-branch.version:1]: @CaseyHillers was this intended to point to the version, or the candidate branch?

Comment (CaseyHillers) [inline bin/internal/release-candidate-branch.version:1]: This is a bug on my part. Can we update this to be flutter-2.13-.candidate.0?

Comment (CaseyHillers) [reply bin/internal/release-candidate-branch.version]: It should point to the branch, so then CP authors can go to something like https://github.com/flutter/flutter/blob/stable/bin/internal/engine.version to find the release branch to send CPs to

Comment (CaseyHillers) [reply bin/internal/release-candidate-branch.version]: I filed https://github.com/flutter/flutter/issues/101773 to track fixing it

Comment (itsjustkevin) [reply bin/internal/release-candidate-branch.version]: I considered writing a bot that would take the human out of the loop and handle this for us, but there seems to be some security implications.

**Review by christopherfujino — APPROVED** [review]
LGTM, also validated binaries are all codesigned.

**Pull request merged**